packages/shared/lib/migration.ts

```diff
@@ -162,47 +163,97 @@ export const createUnsignedBundle = (
-                            seed.set(migrationSeed)
-                            data.set(response.payload)
-                        } else {
-                            data.update((_existingData) =>
-                                Object.assign({}, _existingData, {
-                                    balance: _existingData.balance + response.payload.balance,
-                                    inputs: [..._existingData.inputs, ...response.payload.inputs],
-                                    lastCheckedAddressIndex: response.payload.lastCheckedAddressIndex,
-                                })
-                            )
-                        }
-
-                        prepareBundles()
-
-                        resolve()
-                    },
-                    onError(error) {
-                        reject(error)
-                    },
-                }
-            )
+export const getMigrationData = async (migrationSeed: string, initialAddressIndex = 0): Promise<void> => {
+    if (get(ongoingSnapshot) === true) {
+        openSnapshotPopup()
+    } else {
+        const FIXED_ADDRESSES_GENERATED = 10
+        let totalBalance = 0
+        const inputs: Input[] = []
+
+        for (let index = 0; index < FIXED_ADDRESSES_GENERATED; index++) {
+            const legacyAddress = generateAddress(migrationSeed, index, ADDRESS_SECURITY_LEVEL)
+            const binaryAddress = '0x' + convertToHex(legacyAddress)
+            const balance = await fetchMigratableBalance(binaryAddress)
+
+            totalBalance += balance
+
+            inputs.push({
+                address: legacyAddress,
+                balance,
+                spent: false,
+                index,
+                securityLevel: ADDRESS_SECURITY_LEVEL,
+                spentBundleHashes: [],
+            })
         }
-    })
+
+        const migrationData: MigrationData = {
+            lastCheckedAddressIndex: FIXED_ADDRESSES_GENERATED,
+            balance: totalBalance,
+            inputs: inputs,
+            spentAddresses: false,
+        }
+
+        const { seed, data } = get(migration)
+
+        try {
+            if (initialAddressIndex === 0) {
+                seed.set(migrationSeed)
+                data.set(migrationData)
+            } else {
+                data.update((_existingData) =>
+                    Object.assign({}, _existingData, {
+                        balance: _existingData.balance + migrationData.balance,
+                        inputs: [..._existingData.inputs, ...migrationData.inputs],
+                        lastCheckedAddressIndex: migrationData.lastCheckedAddressIndex,
+                    })
+                )
+            }
+
+            prepareBundles()
+        } catch (error) {
+            console.error(error)
+        }
+    }
+}
+
+async function fetchMigratableBalance(binaryAddress: string): Promise<number> {
+    const BASE_URL = 'https://migrator-api.iota-alphanet.iotaledger.net'
```

Comment:
We need to use the base url based on if its a developer profile or not, developer profile will use this url, but production will use a different one